src/learner.cc

```diff
@@ -319,6 +363,82 @@ class LearnerImpl : public Learner {
       return;
     }
 
+    if (tparam_.model_format == "json") {
+      LOG(WARNING) << "JSON serialization is at experimental stage. "
+                   << "Output schema is subject to change in the future.";
+      Json model { Object() };
+      this->Save(&model);
+      std::stringstream ss;
+      Json::Dump(model, &ss);
+      size_t length = ss.str().size();
```

Comment:
I would actually switch 373 and 374 lines. And write size_t length = str.size()
I am not sure that it is guaranteed that ss.str() consecutive calls return the same object
Otherwise it is just a bit wasteful :)